src/main/java/banking/database/AccountDao.java

```diff
@@ -0,0 +1,101 @@
+package banking.database;
+
+import banking.model.Account;
+import banking.model.Card;
+
+import java.sql.*;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+public class AccountDao {
+
+    private final DBManager dbManager;
+
+    public AccountDao(DBManager dbManager) {
+        this.dbManager = dbManager;
+    }
+
+    public Optional<Account> get(String cardNumber, String cardPIN) {
+        try (Connection connection = dbManager.createConnection();
+             PreparedStatement statement = connection.prepareStatement(
+                     AccountQuery.GET.getQuery()
+             )) {
+            statement.setString(1, cardNumber);
+            statement.setString(2, cardPIN);
+
+            ResultSet rs = statement.executeQuery();
+            if (rs.next()) {
+                return Optional.of(new Account(
+                        rs.getInt("id"),
+                        new Card(rs.getString("cardNumber"), rs.getString("cardPIN")),
+                        rs.getInt("balance")
+                ));
+            }
+        } catch (SQLException e) {
+            e.printStackTrace();
+        }
+        return Optional.empty();
+    }
+
+    public List<Account> getAll() {
+        List<Account> result = new ArrayList<>();
+
+        try (Connection connection = dbManager.createConnection();
+             Statement statement = connection.createStatement()) {
+            ResultSet rs = statement.executeQuery(AccountQuery.GET_ALL.getQuery());
+
+            while (rs.next()) {
+                result.add(new Account(
+                        rs.getInt("id"),
+                        new Card(rs.getString("cardNumber"), rs.getString("cardPIN")),
+                        rs.getInt("balance"))
+                );
+            }
+        } catch (SQLException e) {
+            e.printStackTrace();
```

Comment:
Is empty list our expected behavior for fail cases? Don’t we need to fail whole program?

Reply:
That makes sense, thank you. Added exception throw that is catched in BankingSystem and then stops whole program

Reply:
What about transactions in such cases?

Reply:
Also don’t see any changes

Reply:
I have removed that method as we don't need to receive the account list anymore